supply.go

```diff
@@ -58,6 +58,11 @@ func Supply(values ...interface{}) Option {
 	types := make([]reflect.Type, len(values))
 	for i, value := range values {
 		switch value := value.(type) {
+		case annotated:
```

Comment:
So they look super similar but they're operating on different struct types.
We can't merge them because `value` has a different type in each switch statement.
We can explore deprecating fx.Annotated in favor of fx.Annotate in a future change,
converting fx.Annotated into fx.Annotate before it hits the internals.